File: design-patterns/go/decorator/data_export/json_data_export_decorator.go
// json_data_export_decorator.go

package main

import "fmt"

type JsonDataExportDecorator struct {
	*DataExportDecorator
}

func NewJsonDataExportDecorator(dataExporter DataExport) (jsonDataExportDecorator *JsonDataExportDecorator) {
	jsonDataExportDecorator = &JsonDataExportDecorator{}
	jsonDataExportDecorator.DataExportDecorator = NewDataExportDecorator(dataExporter)
	return
}

func (jsonDataExportDecorator *JsonDataExportDecorator) ProcessData() {
	jsonDataExportDecorator.DataExportDecorator.ProcessData()
	jsonDataExportDecorator.processJson()
}

func (jsonDataExportDecorator *JsonDataExportDecorator) processJson() {
	fmt.Println("Processed data to JSON")
}

File: design-patterns/php/decorator/dataexport/JsonDataExportDecorator.php
<?php
// JsonDataExportDecorator.php

namespace BigBoxCode\DesignPattern\Decorator\DataExport;

class JsonDataExportDecorator extends DataExportDecorator {
    public function __construct(DataExport $dataExporter) {
        parent::__construct($dataExporter);
    }

    public function processData(): void {
        $this->dataExporter->processData();
        $this->processJson();
    }

    public function processJson(): void {
        echo "Processed data to JSON\n";
    }
}